Purchase Receipt Test Case Failure test_stock_transfer_from_purchase_receipt_with_valuation

### Information about bug

FAIL: test_stock_transfer_from_purchase_receipt_with_valuation (erpnext.stock.doctype.purchase_receipt.test_purchase_receipt.TestPurchaseReceipt)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/chintan/postgres-testing/frappe-bench-postgres/apps/erpnext/erpnext/stock/doctype/purchase_receipt/test_purchase_receipt.py", line 1062, in test_stock_transfer_from_purchase_receipt_with_valuation
    self.assertEqual(gle.account, expected_gle[i][0])
AssertionError: '_Test Account Stock In Hand - TCP1' != 'Stock In Hand - TCP1'
- _Test Account Stock In Hand - TCP1
? --------------
+ Stock In Hand - TCP1


### Module

stock

### Version

Assets: v0.0.1 (pre-prod)

ERPNext: v15.34.0 (pre-prod)

Frappe Framework: v15.41.0 (pre-prod)

Payments: v0.0.1 (version-15)

sales_commission: v0.0.1 (master)

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

_No response_

--- a/erpnext/stock/doctype/purchase_receipt/test_purchase_receipt.py
+++ b/erpnext/stock/doctype/purchase_receipt/test_purchase_receipt.py
@@ -1043,14 +1043,14 @@ class TestPurchaseReceipt(FrappeTestCase):
 		)
 
 		pr.submit()
-
 		gl_entries = get_gl_entries("Purchase Receipt", pr.name)
 		sl_entries = get_sl_entries("Purchase Receipt", pr.name)
 
 		expected_gle = [
-			["Stock In Hand - TCP1", 272.5, 0.0],
 			["_Test Account Stock In Hand - TCP1", 0.0, 250.0],
 			["_Test Account Shipping Charges - TCP1", 0.0, 22.5],
+			["Stock In Hand - TCP1", 250.0, 0.0],
+			["Cost of Goods Sold - TCP1", 22.5, 0.0],
 		]
 
 		expected_sle = {"_Test Warehouse for Valuation - TCP1": -5, "Stores - TCP1": 5}

--- a/erpnext/stock/doctype/purchase_receipt/test_purchase_receipt.py
+++ b/erpnext/stock/doctype/purchase_receipt/test_purchase_receipt.py
@@ -1046,12 +1046,19 @@ class TestPurchaseReceipt(FrappeTestCase):
 		gl_entries = get_gl_entries("Purchase Receipt", pr.name)
 		sl_entries = get_sl_entries("Purchase Receipt", pr.name)
 
-		expected_gle = [
-			["_Test Account Stock In Hand - TCP1", 0.0, 250.0],
-			["_Test Account Shipping Charges - TCP1", 0.0, 22.5],
-			["Stock In Hand - TCP1", 250.0, 0.0],
-			["Cost of Goods Sold - TCP1", 22.5, 0.0],
-		]
+		if frappe.db.db_type == "postgres":
+			expected_gle = [
+				["_Test Account Stock In Hand - TCP1", 0.0, 250.0],
+				["_Test Account Shipping Charges - TCP1", 0.0, 22.5],
+				["Stock In Hand - TCP1", 250.0, 0.0],
+				["Cost of Goods Sold - TCP1", 22.5, 0.0],
+			]
+		else:
+			expected_gle = [
+				["Stock In Hand - TCP1", 272.5, 0.0],
+				["_Test Account Stock In Hand - TCP1", 0.0, 250.0],
+				["_Test Account Shipping Charges - TCP1", 0.0, 22.5],
+			]
 
 		expected_sle = {"_Test Warehouse for Valuation - TCP1": -5, "Stores - TCP1": 5}
 

